python/src/poses_bindings.cpp

```diff
@@ -374,7 +477,37 @@ void export_poses()
     {
         MAKE_PTR(CPose3DPDFGaussian)
 
+        // dealing with overloading
+        void (CPose3DPDF::*getCovariance1)(mrpt::math::CMatrixFixedNumeric<double,6ul,6ul> &cov) const = &CPose3DPDFGaussian::getCovariance;
+
         class_<CPose3DPDFGaussian, bases<CPose3DPDF> >("CPose3DPDFGaussian", init<>())
+            .def(init<CPose3DPDFGaussian>())
+            .def(init<CPose3D>())
+            .def("getMean", &CPose3DPDFGaussian::getMean, "Returns the mean, or mathematical expectation of the probability density distribution (PDF).")
+            .def("getCovariance", getCovariance1, "Returns an estimate of the pose covariance matrix (STATE_LENxSTATE_LEN cov matrix).")
+            .def("getCovarianceAndMean", &CPose3DPDFGaussian::getCovarianceAndMean, "Returns an estimate of the pose covariance matrix (STATE_LENxSTATE_LEN cov matrix) and the mean, both at once.")
+            .def("saveToTextFile", &CPose3DPDFGaussian::saveToTextFile, "Save PDF's particles to a text file. See derived classes for more information about the format of generated files.")
+            .def("changeCoordinatesReference", &CPose3DPDF::changeCoordinatesReference, "this = p (+) this. This can be used to convert a PDF from local coordinates to global, providing the point (newReferenceBase) from which 'to project' the current pdf. Result PDF substituted the currently stored one in the object.")
+            .def("getCovarianceEntropy", &CPose3DPDFGaussian::getCovarianceEntropy, "Compute the entropy of the estimated covariance matrix.")
+            .def("drawSingleSample", &CPose3DPDFGaussian::drawSingleSample, "Draws a single sample from the distribution")
+//            .def("copyFrom", &CPose3DPDFGaussian::copyFrom, "Copy operator, translating if necesary (for example, between particles and gaussian representations).")
+            .def("bayesianFusion", &CPose3DPDF::bayesianFusion, "Bayesian fusion of two pose distributions (product of two distributions->new distribution), then save the result in this object (WARNING: See implementing classes to see classes that can and cannot be mixtured!).")
+            .def("inverse", &CPose3DPDFGaussian::inverse, "Returns a new PDF such as: NEW_PDF = (0,0,0) - THIS_PDF")
+            .def("jacobiansPoseComposition", &CPose3DPDFGaussian::jacobiansPoseComposition, "This static method computes the pose composition Jacobians.")
+            .def("__str__", &CPose3DPDFGaussian_asString)
+            .def_readwrite("mean", &CPose3DPDFGaussian::mean)
+            // Note: the following does not work unfortunately: p.cov[35] = 1.0
+            // FIXME: change cov to return actual matrix object (use def_readwrite)
+            .add_property("cov", CPose3DPDFGaussian_get_cov, CPose3DPDFGaussian_set_cov)
```

Comment:
> Is my solution acceptable?

Sure, sounds good. In general, `git rebase origin/master` can be used in such cases. But if master has chnaged significantly, it might give you a lot of conflicts and it might be simpler to manually merge in the additions on the latest master as you suggest.

> instead of using MAKE_PTR macro I switched to MAKE_PTR_BASE macro to setup object inheritance.

Sure, feeld free to update my additions as well to make object inheritance work better. As I said, I know little about idiomatic Boost.Python and just copied what was there or what a quick google search suggested ;-)

> I'll accept this PR, then you can propose yours as a new one, right?

Sounds good, thanks @jlblancoc!